### DISCUSSION TITLE
Contribution Being counted

### DISCUSSION THREAD

**Discussion by hritika2409** [sanjay-kv/Open-source-Practice#966] [Q&A] [answered]
Will my contribution my effects be counted,even if the pa doesnt like it the way it is and doesnt merge it in there project.But i did my 100% regrading the same and worked thought the time period and even tried to change it according to them.

**Comment by lovelymahor** [comment, marked as answer]
Hi hritika2409,

Regarding your question, while it can be disheartening if a Pull Request (PR) isn’t merged, there are several ways in which your contribution is still valuable.

Every contribution, whether merged or not, helps you learn and grow as a developer. The feedback you received and the adjustments you made according to the project administrator's (PA) suggestions are invaluable experiences.

Your contributions are still visible in the commit history and your personal GitHub profile, showcasing your hard work and involvement in open-source projects.

By engaging with the community, asking questions, and working on improvements, you demonstrate your commitment and willingness to collaborate, which is highly regarded in the open-source community.

Your efforts might not have been merged this time, but your perseverance can lead to future contributions being accepted. The knowledge and skills you’ve gained can be applied to other projects or future PRs.

Open source is about collaboration, learning, and sharing. Every contribution, even those not merged, adds to the collective knowledge and progress of the community.

Keep up the great work and continue contributing! Every bit of effort counts and helps you become a better developer.

**Comment by dhruv8433** [comment]
@hritika2409 

Yes, your contributions to a project on GitHub will still be counted even if the pull request (PR) is not merged. Here’s how it works:

Contribution Tracking
1. Commits:
If you create commits on GitHub, they will be counted as contributions on your profile. This includes commits made on branches, whether or not they are eventually merged into the main branch.

2. Pull Requests:
Opening a pull request also counts as a contribution. This is true regardless of whether the pull request is merged, closed, or rejected.

3. Issues:
Creating issues, commenting on issues, and participating in discussions are also tracked as contributions.

4. Profile Activity
Your GitHub profile activity graph (the contribution graph) will reflect the following types of activities:

- Commits to any repository
- Pull requests created
- Issues reported
- Code review comments on pull requests and commits
- Contribution Graph

The green squares on your GitHub contribution graph indicate your activity. Each square represents the number of contributions you’ve made on a particular day, which includes commits, pull requests, issues, and comments.

**Reply by hritika2409** [reply]
but this time my contribution is not counted as per your message i already did these things from day one till now but i m not in the leaderboard
